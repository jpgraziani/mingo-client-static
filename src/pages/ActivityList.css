.activities-list {
  margin: 0;
  padding: 0;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.activity-item {
  flex: 1;
  width: 280px;
  margin-bottom: 1rem;
  border: 2px solid black;
  padding: .5rem;
}