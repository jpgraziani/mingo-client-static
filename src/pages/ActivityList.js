import React from 'react'

import ActivityCard from '../components/ActivityCard'
import ApiContext from '../ApiContext'

export default class ActivityList extends React.Component {

 static contextType = ApiContext;

  render() {
    const { activities=[] } = this.context
    return (
      <section>
        <div>
          <ul>
            {activities.map((activity, idx) => 
              <li key={idx + activity.id}>
                <ActivityCard 
                  id={activity.id}
                  name={activity.name}
                  date={activity.date}
                />
              </li>
            )}
          </ul>
        </div>
      </section>
    );
  }
}