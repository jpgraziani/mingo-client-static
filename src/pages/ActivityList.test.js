import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ActivityList from './ActivityList'

describe(`ActivityList component`, () => {
  it('renders an ActivityList by default', () => {
    const wrapper = shallow(<ActivityList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

