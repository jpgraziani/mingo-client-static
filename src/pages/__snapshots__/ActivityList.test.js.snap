// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList component renders an ActivityList by default 1`] = `
<section
  className="activity-list-container"
>
  <div
    className="act-list-top-nav"
  >
    <Link
      className="add-btn"
      to="/add-activity"
    >
      Add Activity
    </Link>
  </div>
  <div>
    <ul
      className="activities-list"
    />
  </div>
</section>
`;
